exporter/instanaexporter/factory.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instanaexporter
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+
+	instanaConfig "github.com/ibm-observability/instanaexporter/config"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "instana"
+	// The stability level of the exporter.
+	stability = component.StabilityLevelBeta
+)
+
+//NewFactory creates an Instana exporter factory
+func NewFactory() component.ExporterFactory {
+	return component.NewExporterFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithTracesExporter(createTracesExporter, stability),
+	)
+}
+
+// createDefaultConfig creates the default exporter configuration
+func createDefaultConfig() config.Exporter {
+	return &instanaConfig.Config{
+		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
+		HTTPClientSettings: confighttp.HTTPClientSettings{
+			Endpoint: "",
+			Timeout:  30 * time.Second,
+			Headers:  map[string]string{},
+			// We almost read 0 bytes, so no need to tune ReadBufferSize.
+			WriteBufferSize: 512 * 1024,
+		},
+	}
+}
+
+// createTracesExporter creates a trace exporter based on this configuration
+func createTracesExporter(ctx context.Context, set component.ExporterCreateSettings, config config.Exporter) (component.TracesExporter, error) {
+	cfg := config.(*instanaConfig.Config)
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	instanaExporter, err := newInstanaExporter(cfg, set)
+	if err != nil {
+		cancel()
+		return nil, err
+	}
+
+	return exporterhelper.NewTracesExporterWithContext(
+		ctx,
+		set,
+		config,
+		instanaExporter.pushConvertedTraces,
```

Comment:
Where is this one?

Reply:
It is part of the Instana implementation code. This is pushed in the 2nd PR.